src/Jwt/Symfony/Authentication/JsonWebToken.php

```diff
@@ -0,0 +1,141 @@
+        $this->token = (new Parser())->parse($jwt);
+    }
+
+    public function authenticate(): JsonWebToken
+    {
+        return new self($this->getCredentials(), true);
+    }
+
+    public function getUserId(): string
+    {
+        if ($this->token->hasClaim('uid')) {
+            return $this->token->getClaim('uid');
+        }
+
+        if ($this->token->hasClaim('https://publiq.be/uitidv1id')) {
+            return $this->token->getClaim('https://publiq.be/uitidv1id');
+        }
+
+        return $this->token->getClaim('sub');
+    }
+
+    public function getClientId(): ?string
+    {
+        // Check first if the token has the claim, to prevent an OutOfBoundsException (thrown if the default is set to
+        // null and the claim is missing).
+        if ($this->token->hasClaim('azp')) {
+            return (string) $this->token->getClaim('azp');
+        }
+        return null;
+    }
+
+    public function hasClaims(array $names): bool
+    {
+        foreach ($names as $name) {
+            if (!$this->token->hasClaim($name)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public function isUsableAtCurrentTime(): bool
+    {
+        // Use the built-in validation provided by Lcobucci without any extra validation data.
+        // This will automatically validate the time-sensitive claims.
+        // Set the leeway to 30 seconds so we can compensate for slight clock skew between auth0 and our own servers.
+        // @see https://self-issued.info/docs/draft-ietf-oauth-json-web-token.html#nbfDef
+        return $this->token->validate(new ValidationData(null, self::TIME_LEEWAY));
+    }
+
+    public function hasValidIssuer(array $validIssuers): bool
+    {
+        return in_array($this->token->getClaim('iss', ''), $validIssuers, true);
+    }
+
+    public function hasAudience(string $audience): bool
+    {
+        if (!$this->token->hasClaim('aud')) {
+            return false;
+        }
+
+        // The aud claim can be a string or an array. Convert string to array with one value for consistency.
+        $aud = $this->token->getClaim('aud');
+        if (is_string($aud)) {
+            $aud = [$aud];
+        }
+
+        return in_array($audience, $aud, true);
+    }
+
+    public function hasEntryApiInPubliqApisClaim(): bool
+    {
+        $apis = $this->token->getClaim('https://publiq.be/publiq-apis', '');
+
+        if (!is_string($apis)) {
+            return false;
+        }
+
+        $apis = explode(' ', $apis);
+        return in_array('entry', $apis, true);
+    }
+
+    public function verifyRsaSha256Signature(string $publicKey, ?string $keyPassphrase = null): bool
+    {
+        $signer = new Sha256();
+        $key = new Key($publicKey, $keyPassphrase);
+        return $this->token->verify($signer, $key);
+    }
+
+    public function getCredentials(): string
+    {
+        return $this->jwt;
+    }
+
+    public function toSentryTags(): array
```

Comment:
I wonder if this is required here? It also seems to be used only once inside `SentryHandlerScopeDecorator`

Reply:
Yeah I get what you mean. I wasn't sure, but I was trying to not "leak" the `Token` class used internally into the rest of the application so the app only knows about `JsonWebToken` and doesn't use the `Token` class directly somewhere. But then I either need to add a generic `getClaim()` method on `JsonWebToken`, which I also wanted to avoid so we have a very specific interface and more control over what claims are being read. So the only other solution I thought of was adding methods like `getUidClaim()`, `getUiTIDv1IdClaim()`, and `getSubClaim()`, but then that might get confusing with `getUserId()` which should be used normally _except_ in the Sentry tags?

Reply:
Good point! Then it's indeed much better to keep `toSentryTags` method inside `JsonWebToken`

Reply:
Ah I think I've found another potential solution, I'll push it so you can see what you think